FWCore/Framework/interface/Schedule.h

```diff
@@ -277,6 +277,8 @@ namespace edm {
     /// Deletes module with label iLabel
     void deleteModule(std::string const& iLabel, ActivityRegistry* areg);
 
+    void initializeEarlyDelete(std::vector<std::string> branchesToDeleteEarly, edm::ProductRegistry const& preg);
```

Comment:
I do not like the use of passing by value here. The internals of the code do not care about it. I understand that when called, the variable is supposed to be cleared, but that is more a matter of the call sight understanding the lifetime of the variable and NOT a property of this function.

Reply:
Yeah, that's probably better way. I was originally hoping to avoid making copies of the vector (that gets modified in `initializeBranchToReadingWorker()`), but since the vector (currently) needs to be copied for each `StreamSchedule`, it wouldn't change anything for the copies to pass the `vector` here by reference-to-`const` and make the copy in `initializeBranchToReadingWorker()`.